src/main/java/org/galatea/starter/entrypoint/IexRestController.java

```diff
@@ -48,4 +64,52 @@ public List<IexLastTradedPrice> getLastTradedPrice(
     return iexService.getLastTradedPriceForSymbols(symbols);
   }
 
+  /**
+   * Get the historical prices of specified symbols.
+   *
+   * @param symbols list of symbols to get historical price for.
+   * @param range Optional Parameter specifying the Range of time that the price should be
+   *     measured over
+   * @param date Optional Parameter specifying the exact date the price should be measured on
+   * @return a IexLastTradedPrice object for the given symbol.
+   */
+  @GetMapping(value = "${mvc.iex.getHistoricalPricePath}", produces = {
+      MediaType.APPLICATION_JSON_VALUE})
+  public IexHistoricalPrice getHistoricalPrice(
+      @RequestParam(value = "symbol") @NotNull final String symbols,
+      @RequestParam(value = "range", required = false) final Optional<@Pattern.List(
+          {@Pattern(regexp = ".*"), @Pattern(regexp = VALID_DATES)}) String> range,
+      @RequestParam(value = "date", required = false) final Optional<Integer> date) {
+    IexHistoricalPrice price;
+//@Pattern.List({@Pattern(regexp =".*"), @Pattern (regexp = VALID_DATES)})
+    //3 Parameters: Symbol is REQUIRED, Date/Range are optional and cannot both be present
+
+    if (date.isPresent()) {
+      int dateVal = date.get();
+
+      //Max validation depends on current time:
+      if (dateVal >= System.currentTimeMillis()) {
```

Comment:
`date` in the request is a `yyyyMMdd` value like `20221017`, right? Comparing it to System.currentTimeMillis() wouldn't provide any meaningful insight, since currentTimeMillis() returns a [Unix timestamp](https://en.wikipedia.org/wiki/Unix_time). A more appropriate comparison would be something closer to the fiveYearsAgo check you have below.